### PR TITLE
ChatArea refactor

### PR DESCRIPTION
**Pull request by 4z0t**
Most of lobby issues come from chat: in some cases a lots of messages can cause lobby bug or even crash and here is why. 
Every message you receive creates a control. More messages = more controls = more usage of memory and, most importantly, when you Alt-Tab each LazyVar calculates its position on frame, which are 6 times more than controls...

### CODE REVIEW

File: lua/ui/lobby/chatarea.lua [hunk 1]
```diff
@@ -3,255 +3,213 @@ local Text = import('/lua/maui/text.lua')
...
+        self.Style = defaultStyle
 
-        -- iInitial width and height are necessary to dodge partial-initialization/re-flow weirdness.
         self.Width:Set(width)
         self.Height:Set(height)
 
-        -- re-flow chat lines when the width is changed.
-        self.Width.OnDirty = function()
+        self.Width.OnDirty = function(lazyvar)
             LOG('chatArea dirty Width')
```
Comment (Garanas) [inline lua/ui/lobby/chatarea.lua:43]: There's a LOG statement here

File: lua/ui/lobby/chatarea.lua [hunk 2]
```diff
@@ -3,255 +3,213 @@ local Text = import('/lua/maui/text.lua')
...
             self:ReflowLines()
         end
-        self.Height.OnDirty = function()
+        self.Height.OnDirty = function(lazyvar)
             LOG('chatArea dirty Height')
```
Comment (Garanas) [inline lua/ui/lobby/chatarea.lua:47]: Same here

File: lua/ui/lobby/chatarea.lua [hunk 3]
```diff
@@ -3,255 +3,213 @@ local Text = import('/lua/maui/text.lua')
...
+local defaultStyle = {
+    message = {
+        fontColor = LazyVar.Create('FFFFFFFF'),
+        fontFamily = LazyVar.Create(UIUtil.bodyFont),
+    },
+    author = {
+        fontColor = LazyVar.Create('FFFFFFFF'),
+        fontFamily = LazyVar.Create("Arial Gras"),
+    },
+    fontSize = LazyVar.Create(tonumber(Prefs.GetFromCurrentProfile('LobbyChatFontSize')) or 14),
```
Comment (Garanas) [inline lua/ui/lobby/chatarea.lua:17]: what if this returns nil, how does `tonumber` respond to that?

File: lua/ui/lobby/chatarea.lua [hunk 4]
```diff
@@ -3,255 +3,213 @@ local Text = import('/lua/maui/text.lua')
 local UIUtil = import('/lua/ui/uiutil.lua')
 local LayoutHelpers = import('/lua/maui/layouthelpers.lua')
 local Prefs = import('/lua/user/prefs.lua')
-
---- Represents a control that keep tracks and displays chat messages and their author's names
-ChatArea = Class(Group){
+local LazyVar = import('/lua/lazyvar.lua')
+
+local defaultStyle = {
```
Comment (Garanas) [inline lua/ui/lobby/chatarea.lua:8]: nice caching 😄

**Comment by Garanas** [comment]
Looks good - nice feature to pm people on clicking their name.

File: lua/ui/lobby/chatarea.lua (hunk 3 above)
Comment (4z0t) [reply lua/ui/lobby/chatarea.lua:17]: Everything that represents number is a number, otherwise nil

**Comment by 4z0t** [comment]
Haven't seen that those logs even appear at all, but there were before me, so, let them be I think 😁

**Comment by 4z0t** [comment]
I wanted to sync somehow players' color in chat, what do you think @Garanas ?

**Comment by Garanas** [comment]
Let us discuss that on Discord in #game-repository and work on that in a separate pull request.

**Comment by Garanas** [comment]
I'm not available the next five days - if anything critical breaks because of this then feel free to merge a fix.

**Pull request merged**